frontend/src/app/App.test.tsx

```diff
@@ -230,311 +239,382 @@ describe("App", () => {
-          currentState: getHostCommunicationState({
-            cacheClearRequested: true,
-          }),
-        }),
-      })
-    )
-    expect(instance.props.hostCommunication.onCacheClear).toHaveBeenCalled()
-  })
+    it("changes scriptRunState and triggers rerunScript when scriptRerunRequested message has been received", () => {
+      instance.isServerConnected = jest.fn().mockReturnValue(true)
 
-  it("does not prevent a modal from opening when closure message is set", () => {
-    const onModalReset = jest.fn()
-    const wrapper = shallow(
-      <App
-        {...getProps({
-          hostCommunication: getHostCommunicationProp({
-            currentState: getHostCommunicationState({
-              menuItems: [],
-              queryParams: "",
-              forcedModalClose: false,
-            }),
-            onModalReset,
-            sendMessage: jest.fn(),
-          }),
-        })}
-      />
-    )
-    wrapper.setProps(
-      getProps({
-        hostCommunication: getHostCommunicationProp({
-          currentState: getHostCommunicationState({ forcedModalClose: true }),
-          onModalReset,
-        }),
-      })
-    )
-    expect(onModalReset).toBeCalled()
-    wrapper.setProps(
-      getProps({
-        hostCommunication: getHostCommunicationProp({
-          currentState: getHostCommunicationState({ forcedModalClose: false }),
-          onModalReset,
-        }),
-      })
-    )
-    const dialog = StreamlitDialog({
-      type: DialogType.ABOUT,
-      sessionInfo: mockSessionInfo(),
-      onClose: () => {},
+      const rerunScriptFunc = jest.spyOn(
+        // @ts-expect-error
+        instance.hostCommunicationMgr.props,
+        "rerunScript"
+      )
+
+      dispatchEvent(
+        "message",
+        new MessageEvent("message", {
+          data: {
+            stCommVersion: HOST_COMM_VERSION,
+            type: "RERUN_SCRIPT",
+          },
+          origin: "https://devel.streamlit.test",
+        })
+      )
+
+      expect(rerunScriptFunc).toHaveBeenCalledWith()
+      expect(wrapper.state("scriptRunState")).toBe(
+        ScriptRunState.RERUN_REQUESTED
+      )
     })
-    wrapper.setState({ dialog })
-    expect(wrapper.find(Modal)).toHaveLength(1)
-  })
 
-  it("sends SCRIPT_RUN_STATE_CHANGED signal to the host when the app is first rendered", () => {
-    const props = getProps()
-    shallow(<App {...props} />)
+    it("fires clearCache function when cacheClearRequested message has been received", () => {
+      instance.isServerConnected = jest.fn().mockReturnValue(true)
 
-    expect(props.hostCommunication.sendMessage).toHaveBeenCalledWith({
-      type: "SCRIPT_RUN_STATE_CHANGED",
-      scriptRunState: ScriptRunState.NOT_RUNNING,
+      const clearCacheFunc = jest.spyOn(
+        // @ts-expect-error
+        instance.hostCommunicationMgr.props,
+        "clearCache"
+      )
+      const closeDialogFunc = jest.spyOn(instance, "closeDialog")
```

Comment:
nit: may be worth mentioning that `clearCache` calls this method in a comment since it's not entirely clear what this code is for from the test alone (it'd also probably be fine to just not test for this since the fact that `clearCache` calls `closeDialog` should ideally be tested in a separate unit test and not by the hostCommunication test)